### ISSUE TITLE
508-defect-2 [SCREENREADER]: Search fields should have the landmark role="search"

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-cms#13138]
## User Story or Problem Statement

As a screen reader user, I expect search fields on the page to be discoverable using my landmarks list.

## Details

> Landmarks can be used by assistive technology to quickly identify and navigate to large sections of the document. The search role is added to the container element that encompasses the items and objects that, as a whole, combine to create search functionality. When a <form> is a search form, use the search role instead of form. If a document includes more than one search, each should have a unique label. While there is an input of type search, there is no HTML element that defines a search landmark.

[MDN web docs on the search role.](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Search_role)

All search field forms on the page should have a child `div` with `role="search"` to help screen reader users easily find or track back to the search field using their landmarks or regions list. Although MDN recommends attaching `role="search"` to the `form` element itself; there still appears to be issues with VoiceOver where it won't emerge as a landmark. Using a child `div` instead is a [best practice recommended by Adrian Roselli](https://adrianroselli.com/2015/08/where-to-put-your-search-role.html) which resolves the issue.

For example:
``` html
<form>
  <!-- div with role="search" -->
  <div role="search" aria-label="descriptive label here">
    <!-- search fields here -->
  </div>
</form>
```

The mobile resources search results page is the one instance where this _isn't_ possible, which would break consistency for mobile screen reader users.

<img width="1435" alt="Screen Shot 2021-06-21 at 3 07 49 PM" src="https://user-images.githubusercontent.com/14154792/122814654-81413000-d2a2-11eb-9771-0d1ec12548bb.png">

Because this is a `button` that _contains_ the search form, we can't override its role into `search`. That means that the search _won't_ be discoverable to mobile screen reader users when navigating by landmarks or regions. More research is needed.

## Acceptance Criteria

- [x] All search `form`s should have a child `div` with `role="search"` that wraps around all search fields
- [x] In cases of multiple search fields on the page, a unique `aria-label` is provided
- [x] Find a Form: replace search with `<va-search-input>`component, and aria-label and verify form works as expected
- [x] a11y consult: verify with @laflannery prior to merge
- [ ] Design review: since using new component, verify visual presentation with @wesrowe @thejordanwood 
    - [ ] If any visual differences: verify with Design System team re: usage

## Before

https://user-images.githubusercontent.com/14154792/122939199-6f629a00-d341-11eb-9d15-60c2a07498bd.mov

## After

https://user-images.githubusercontent.com/14154792/122939220-738eb780-d341-11eb-88a5-238aa357582d.mov

**Comment by jilladams** [comment]
@laflannery from planning, DaveC agrees that we'll swap the component. Can you advise re: preferred aria label to use here?

**Comment by laflannery** [comment]
`aria-label` should be "Search VA form" - The screenshot of the markup above (pasted again below) was a markup of what this would look like when the va component was implemented - included the `<div>` and `aria-label`. Apologies for not making that clear in my previous comment

![image](https://user-images.githubusercontent.com/106678594/229202823-9dc11d6d-288f-48a7-b319-039582fcab50.png)

**Comment by dsasser** [comment]
# End of Sprint Status Update 5/23/23

Work is complete, waiting on QA. QA is blocked due to Review Instance failures, which Platform is working on. See https://dsva.slack.com/archives/CBU0KDSB1/p1684769769244749.

**Comment by laflannery** [comment]
@dsasser please make sure I am added as a reviewer on the PR so I an take a look before it is merged Thanks!

**Comment by dsasser** [comment]
> @dsasser please make sure I am added as a reviewer on the PR so I can take a look before it is merged Thanks!

The plan was to do this once the review instance was up and running. Unfortunately, Platform is having trouble with review instances of latee.

**Comment by dsasser** [comment]
# Status Update 5/26/23

We remain blocked on QA due to [Review Instance issues Platform is working on](https://dsva.slack.com/archives/CBU0KDSB1/p1684767619143429).
